### PR TITLE
Lexer: add non-breaking spaces to whitespace

### PR DESCRIPTION
**Pull request by jpers36**
The lexer can't lex non-breaking spaces.  This PR will add non-breaking spaces to the whitespace lexer.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_ansi.py [hunk 1]
```diff
@@ -58,7 +58,7 @@
 
 ansi_dialect.set_lexer_matchers(
     [
-        RegexLexer("whitespace", r"[\t ]+", WhitespaceSegment),
+        RegexLexer("whitespace", r"[\t\xc2\xa0 ]+", WhitespaceSegment),
```
Comment (jpy-git) [inline src/sqlfluff/dialects/dialect_ansi.py:61]: @jpers36 I don't doubt this works but feels a bit inelegant. Doing a bit of research I saw a comment that the regex `\s` also matches non-breaking spaces, however it also matches newlines.

A good post [here](https://stackoverflow.com/questions/3469080/match-whitespace-but-not-newlines) suggests we can use this double negative regex pattern `[^\S\r\n]` to do `\s` but not newlines.
Could you try that please and see if that also solves the issue?

Comment (jpers36) [reply src/sqlfluff/dialects/dialect_ansi.py]: @jpy-git that works, thanks!

Comment (jpy-git) [reply src/sqlfluff/dialects/dialect_ansi.py]: Nice!

File: src/sqlfluff/dialects/dialect_ansi.py [hunk 2]
```diff
@@ -58,7 +58,7 @@
 
 ansi_dialect.set_lexer_matchers(
     [
-        RegexLexer("whitespace", r"[\t ]+", WhitespaceSegment),
+        RegexLexer("whitespace", r"[^\S\r\n]+", WhitespaceSegment),
```
Comment (jpy-git) [inline src/sqlfluff/dialects/dialect_ansi.py:61]: Could we also add a comment here to the effect of:
```
# Match all forms of whitespace except newlines and carriage returns:
# https://stackoverflow.com/questions/3469080/match-whitespace-but-not-newlines
# This pattern allows us to also match non-breaking spaces (#2189).
```
so people don't question our choice. 😄  

Also the whitespace pattern gets used in a couple different places so we should update these too:
https://github.com/sqlfluff/sqlfluff/blob/a5378224e2f1d0441137645179dfeb076dccc642/src/sqlfluff/dialects/dialect_ansi.py#L79

https://github.com/sqlfluff/sqlfluff/blob/a5378224e2f1d0441137645179dfeb076dccc642/src/sqlfluff/dialects/dialect_tsql.py#L123

https://github.com/sqlfluff/sqlfluff/blob/a5378224e2f1d0441137645179dfeb076dccc642/test/core/parser/lexer_test.py#L86

https://github.com/sqlfluff/sqlfluff/blob/a5378224e2f1d0441137645179dfeb076dccc642/test/core/parser/lexer_test.py#L88

**Review by jpy-git — COMMENTED** [review]
@jpers36 Couple more places to update and add the comment, but otherwise looks good 👍

File: src/sqlfluff/dialects/dialect_ansi.py (hunk 2 above)
Comment (jpers36) [reply src/sqlfluff/dialects/dialect_ansi.py:64]: @jpy-git  got them!

**Review by jpy-git — APPROVED** [review]
@jpers36 looks great, thanks for adding this! LGTM 🚀

**Pull request merged**